internal/parser/parser.go

```diff
@@ -1336,26 +1336,6 @@ func (p *parser) consume() {
 // token. If a token has been buffered by peek, then the token will
 // be read if it matches or will be discarded if it is the wrong token.
 func (p *parser) expect(exp token.Token) (token.Pos, string) {
-	if p.buffered {
-		p.buffered = false
-		if p.tok == exp || p.tok == token.EOF {
-			if p.tok == token.EOF {
-				p.errs = append(p.errs, ast.Error{
-					Msg: fmt.Sprintf("expected %s, got EOF", exp),
-				})
-			}
-			return p.pos, p.lit
-		}
-		p.errs = append(p.errs, ast.Error{
-			Msg: fmt.Sprintf("expected %s, got %s (%q) at %s",
-				exp,
-				p.tok,
-				p.lit,
-				p.s.File().Position(p.pos),
-			),
-		})
-	}
-
```

Comment:
Why has this been removed?

Reply:
It was redundant code, in porting the Go code to Rust I saw that it was not necessary. This was confirmed after removing it and all tests still pass.

In short the logic the check for `buffered` is already inside the call to `p.scan()` and the error logic is identical to the logic below after calling scan.